pkg/providers/cloudstack/cloudstack.go

```diff
@@ -87,6 +94,34 @@ func (p *cloudstackProvider) PostWorkloadInit(ctx context.Context, cluster *type
 }
 
 func (p *cloudstackProvider) UpdateSecrets(ctx context.Context, cluster *types.Cluster) error {
+	var contents bytes.Buffer
+	err := p.createSecrets(ctx, cluster, &contents)
+	if err != nil {
+		return err
+	}
+
+	err = p.providerKubectlClient.ApplyKubeSpecFromBytes(ctx, cluster, contents.Bytes())
```

Comment:
We need more validation here - see https://github.com/aws/eks-anywhere/blob/main/designs/cloudstack-multiple-endpoints.md#cloudstack-credentials. Basically we want to make these secrets write-once so they can't be updated in this code. If the secret already exists on the cluster, we should not modify it in any way. Further, we should validate that the contents of the secret are the same as the contents of the secret locally generated to fail fast if anyone is trying to change them.

Reply:
Who is allowed to modify these secrets then, and at what point?

Reply:
I changed the caller hook calling UpdateSecrets function to PostBootstrapSetup. As far as I know PostBootstrapSetup is called after creating a bootstrap cluster on creating a management cluster. In this way, the secrets are created only once when creating a management cluster using the eks-a CLI. The operator of the management cluster can modify the secret directly using the kubeconfig of the cluster if they want to update the credentials of a certain ACS.